src/Kiota.Builder/Writers/TypeScript/CodeMethodWriter.cs

```diff
@@ -362,30 +356,34 @@ private void WriteRequestGeneratorBody(CodeMethod codeElement, RequestParams req
+                ComposeContentInRequestGeneratorBody(requestParams.requestBody, requestAdapterProperty, codeElement.RequestBodyContentType, writer);
             }
         }
 
         writer.WriteLine($"return {RequestInfoVarName};");
     }
-    private static string GetPropertyCall(CodeProperty property, string defaultValue) => property == null ? defaultValue : $"this.{property.Name}";
-    private void WriteSerializerBody(bool inherits, CodeClass parentClass, LanguageWriter writer)
+
+    private void ComposeContentInRequestGeneratorBody(CodeParameter requestBody, CodeProperty requestAdapterProperty, string contentType, LanguageWriter writer)
     {
-        var additionalDataProperty = parentClass.GetPropertyOfKind(CodePropertyKind.AdditionalData);
-        if (inherits)
-            writer.WriteLine("super.serialize(writer);");
-        foreach (var otherProp in parentClass.GetPropertiesOfKind(CodePropertyKind.Custom).Where(static x => !x.ExistsInBaseType && !x.ReadOnly))
+        if (requestBody.Type.Name.Equals(localConventions?.StreamTypeName, StringComparison.OrdinalIgnoreCase))
+        {
+            writer.WriteLine($"{RequestInfoVarName}.setStreamContent({requestBody.Name});");
+            return;
+        }
+
+        var spreadOperator = requestBody.Type.AllTypes.First().IsCollection ? "..." : string.Empty;
+        if (requestBody.Type is CodeType currentType && currentType.TypeDefinition is CodeInterface codeInterface)
+        {
+            var serializerName = $"serialize{codeInterface.Name.ToFirstCharacterUpperCase()}";
+            writer.WriteLine($"{RequestInfoVarName}.setContentFromParsable(this.{requestAdapterProperty.Name.ToFirstCharacterLowerCase()}, \"{contentType}\", body as any, {serializerName});");
+        }
+        else
         {
-            var isCollectionOfEnum = otherProp.Type is CodeType cType && cType.IsCollection && cType.TypeDefinition is CodeEnum;
-            var spreadOperator = isCollectionOfEnum ? "..." : string.Empty;
-            var otherPropName = otherProp.Name.ToFirstCharacterLowerCase();
-            var undefinedPrefix = isCollectionOfEnum ? $"this.{otherPropName} && " : string.Empty;
-            writer.WriteLine($"{undefinedPrefix}writer.{GetSerializationMethodName(otherProp.Type, parentClass, writer)}(\"{otherProp.WireName}\", {spreadOperator}this.{otherPropName});");
+            writer.WriteLine($"{RequestInfoVarName}.setContentFromScalar(this.{requestAdapterProperty.Name.ToFirstCharacterLowerCase()}, \"{contentType}\", {spreadOperator}{requestBody.Name});");
         }
-        if (additionalDataProperty != null)
-            writer.WriteLine($"writer.writeAdditionalData(this.{additionalDataProperty.Name.ToFirstCharacterLowerCase()});");
     }
+    private static string GetPropertyCall(CodeProperty property, string defaultValue) => property == null ? defaultValue : $"this.{property.Name}";
```

Comment:
It looks like `property` will never be null here. All usages pass a non null property and the parameter should probably have the nullable annotation as `CodeProperty?` if this is possible.